pkg/codegen/go/gen.go

```diff
@@ -5246,6 +5246,8 @@ func Pkg%[1]sDefaultOpts(opts []pulumi.%[1]sOption) []pulumi.%[1]sOption {
 		if info.(GoPackageInfo).RespectSchemaVersion && pkg.pkg.Version() != nil {
 			versionPackageRef = fmt.Sprintf("semver.MustParse(%q)", p.Version.String())
 		}
+	} else if pkg.pkg.SupportPack() && pkg.pkg.Version() != nil {
+		versionPackageRef = fmt.Sprintf("semver.MustParse(%q)", p.Version.String())
 	}
```

Comment:
The actual fix

Reply:
If SupportPack is true then we should error if Version is nil

Reply:
That should probably happen during schema bind though, not sdk-gen, I can open another PR for that

Reply:
It probably should, I think I've done err checks in all the other sdk gens though. Happy for you to clean them up and defer to bind doing the right thing.

Reply:
I haven't looked at all validations yet but for this one here https://github.com/pulumi/pulumi/pull/17420